# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsDNSPrefetch.h
# Commit: 1a38b75a7dbb
# Full Hash: 1a38b75a7dbb86fa14d2763e7a983dfd0bde5d50
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2021-01-22 10:17:55
# Regressor Bug: 1686828
# File Overlap Count: 1
# Description:
#   Bug 1686828 - Use https OriginAttributes to fetch HTTPS RR r=necko-reviewers,timhuang,dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102234
# ==============================================================================

diff -r e83b6fb52b62 -r 1a38b75a7dbb netwerk/base/nsDNSPrefetch.h
--- a/netwerk/base/nsDNSPrefetch.h	Thu Jan 21 22:08:27 2021 +0000
+++ b/netwerk/base/nsDNSPrefetch.h	Thu Jan 21 22:08:26 2021 +0000
@@ -31,6 +31,9 @@
   nsDNSPrefetch(nsIURI* aURI, mozilla::OriginAttributes& aOriginAttributes,
                 nsIRequest::TRRMode aTRRMode, nsIDNSListener* aListener,
                 bool storeTiming);
+  // For fetching HTTPS RR.
+  nsDNSPrefetch(nsIURI* aURI, mozilla::OriginAttributes& aOriginAttributes,
+                nsIRequest::TRRMode aTRRMode);
   bool TimingsValid() const {
     return !mStartTimestamp.IsNull() && !mEndTimestamp.IsNull();
   }
@@ -47,11 +50,11 @@
   nsresult PrefetchLow(bool refreshDNS = false);
 
   nsresult FetchHTTPSSVC(
-      bool aRefreshDNS, std::function<void(nsIDNSHTTPSSVCRecord*)>&& aCallback);
+      bool aRefreshDNS, bool aPrefetch,
+      std::function<void(nsIDNSHTTPSSVCRecord*)>&& aCallback);
 
  private:
   nsCString mHostname;
-  bool mIsHttps;
   mozilla::OriginAttributes mOriginAttributes;
   bool mStoreTiming;
   nsIRequest::TRRMode mTRRMode;